### PR TITLE
feat: add metrics for robustness in binary classification

### PR DESCRIPTION
**Pull request by fracarfer5**
- Adding robustness metrics and attackers
- Adding documentation
- Adding tests
- Adding tutorials

### CODE REVIEW

File: tests/robustness/attackers/test_classification_attackers.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    y_test = test_y
+
+    model = LogisticRegression()
+    model.fit(X_train, y_train)
+    y_pred = model.predict(X_test)
+    baseline_accuracy = accuracy_score(y_test, y_pred)
+
+    proxy = BinaryClassificationProxy(predict=model.predict, predict_proba=model.predict_proba, classes=[0, 1])
+
+    predict_proba_fn = format_function_predict_proba(proxy.learning_task, proxy.predict_proba)  # type: ignore
```
Comment (crismunoz) [inline tests/robustness/attackers/test_classification_attackers.py]: proxy object must be passed to ZooAttack. format_function_predict_proba must be an internal function

File: tests/robustness/attackers/test_classification_attackers.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    dataset = load_dataset("adult", protected_attribute='race') # x,y,group_a,group_b
+    dataset = dataset.groupby(['y','group_a']).sample(SHARD_SIZE, random_state=42) # 0-ga | 0-gb  | 1-ga | 1-gb
+    train_test = dataset.train_test_split(test_size=0.02, stratify=dataset['y'], random_state=0)
+    train = train_test['train']
+    test = train_test['test']
+    correlations = train['X'].corrwith(train['y']).sort_values(ascending=False)
+    top_10_features = correlations.head(10).index.tolist()
+    train['X'] = train['X'][top_10_features]
+    test['X'] = test['X'][top_10_features]
+    return train['X'], test['X'], train['y'], test['y']
```
Comment (crismunoz) [inline tests/robustness/attackers/test_classification_attackers.py]: return train , test

File: tests/robustness/attackers/test_classification_attackers.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+
+def test_hsj(categorical_dataset):
+    train_X, test_X, train_y, test_y = categorical_dataset
+
+    # Standardize data and fit model
+    scaler = StandardScaler()
+    X_train = scaler.fit_transform(train_X)
```
Comment (crismunoz) [inline tests/robustness/attackers/test_classification_attackers.py]: X_train = scaler.fit_transform(train["X"]) the same for test

File: tests/robustness/attackers/test_classification_attackers.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+    X_test = scaler.transform(test_X)
+    feature_names = list(test_X.columns)
+
+    y_train = train_y
```
Comment (crismunoz) [inline tests/robustness/attackers/test_classification_attackers.py]: the same for y train["y'"]

File: tests/robustness/attackers/test_classification_attackers.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    y_pred = model.predict(X_test)
+    baseline_accuracy = accuracy_score(y_test, y_pred)
+
+    proxy = BinaryClassificationProxy(predict=model.predict, predict_proba=model.predict_proba, classes=[0, 1])
+
+    predict_proba_fn = format_function_predict_proba(proxy.learning_task, proxy.predict_proba)  # type: ignore
+
+    zoo_attacker = ZooAttack(name="Zoo", predict_proba_fn=predict_proba_fn)
+
+    zoo_adv_x = zoo_attacker.generate(pd.DataFrame(X_test, columns=feature_names))
```
Comment (crismunoz) [inline tests/robustness/attackers/test_classification_attackers.py]: remove  pd.DataFrame

**Review by crismunoz — CHANGES_REQUESTED** [review]

File: tests/robustness/attackers/test_classification_attackers.py (hunk 5 above)
Comment (fracarfer5) [reply tests/robustness/attackers/test_classification_attackers.py]: done!

File: tests/robustness/attackers/test_classification_attackers.py (hunk 4 above)
Comment (fracarfer5) [reply tests/robustness/attackers/test_classification_attackers.py]: updated with pipeline

File: tests/robustness/attackers/test_classification_attackers.py (hunk 3 above)
Comment (fracarfer5) [reply tests/robustness/attackers/test_classification_attackers.py]: updated with pipeline

File: tests/robustness/attackers/test_classification_attackers.py (hunk 2 above)
Comment (fracarfer5) [reply tests/robustness/attackers/test_classification_attackers.py]: done

File: tests/robustness/attackers/test_classification_attackers.py (hunk 1 above)
Comment (fracarfer5) [reply tests/robustness/attackers/test_classification_attackers.py]: updated

**Review by crismunoz — DISMISSED** [review]
nice!

**Pull request merged**